### DISCUSSION TITLE
How can I make customized permutation problem?

### DISCUSSION THREAD

**Discussion by skyblue0807** [anyoptimization/pymoo#524] [Q&A]
Hello, all. 

I hope to solve permutation problem with GA in pymoo, but in PERMUTATION page, there are only examples using "create_random_TSP/flowshop" function.

I have elements to be permutated, and object functions to evaluate permuated elements list.

How can I build my custom problem in pymoo?
what type of problem class should I use?

**Comment by blankjul** [comment]
Just check out how the TSP problem is implemented in pymoo itself. You can find the source code here: 
https://github.com/anyoptimization/pymoo/blob/main/pymoo/problems/single/traveling_salesman.py

For a custom problem also check out the getting started guide:
https://pymoo.org/getting_started/part_2.html
